### DISCUSSION TITLE
yolo-obb

### DISCUSSION THREAD

**Discussion by hellotenet** [ultralytics/ultralytics#12735] [Q&A]
Excuse me, when using a self-trained YOLO-OBB for prediction, is it possible to return a TXT document in DOTA format for the labels?

**Comment by pderrenger** [comment]
@hellotenet hello! Currently, YOLOv8 does not directly support exporting prediction results in DOTA format as a TXT file. However, you can easily write a script to convert the results from the detection to DOTA format. After running predictions and receiving your results object, you can format and save this data as needed. Here's a brief Python snippet to guide you:

```python
# Assuming 'results' is your prediction output
for result in results:
    with open('output.txt', 'w') as f:
        for box in result.obb.xyxyxyxy:  # using oriented bounding boxes
            label = result.names[int(box[-1])]
            coordinates = list(map(int, box[:-1]))
            f.write(f"{label} " + " ".join(map(str, coordinates)) + "\n")
```

This snippet demonstrates saving the bounding box coordinates and labels to a TXT file. Adjust it as necessary to match the precise DOTA format requirements.
